tests/helpers/testutils.py

```diff
@@ -193,7 +193,7 @@ def pattern_match(*, pattern, value):
 def abs_datapath():
     """Get the absolute path to the end2end data directory."""
     file_abs = os.path.abspath(os.path.dirname(__file__))
```

Comment:
I suppose this `os.path.abspath` isn't needed anymore if you call it at the end. Also should probably change the variable name.

But if you don't mind, I'd rewrite this to use pathlib anyways, using something like:

```python
path = pathlib.Path(__file__).parents[1] / "end2end" / "data"
return path.resolve()
```

or somesuch.